app/main.py

```diff
@@ -1 +1,41 @@
-#  write your code here
+class Animal:
+
+    def __init__(self, name: str, appetite: int,
+                 is_hungry: bool = True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self: str) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self: int) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+        return 0
+
+
+class Cat(Animal):
+
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=3, is_hungry=is_hungry)
+
+    def catch_mouse(self) -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry: bool = True) -> None:
+        super().__init__(name, appetite=7, is_hungry=is_hungry)
+
+    def bring_slippers(self: str) -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list[Animal]) -> int:
+    total__food_points = 0
+    for animal in animals:
+        total__food_points += animal.feed()
+    return total__food_points
```

Comment:
This implementation violates checklist item https://github.com/mate-academy/py-feed-animals/pull/2: "Don't use a for loop if you can use generator expression." Please refactor this function to use sum() with a generator expression for a more concise solution, as shown in the checklist example.